lib/events/filesessions/fileasync_test.go

```diff
@@ -659,25 +660,31 @@ func emitStream(ctx context.Context, t *testing.T, streamer events.Streamer, inE
 }
 
 // readStream reads and decodes the audit stream from uploadID
+// Additionally, readStream will remove any duplicate audit events (by index) encountered.
 func readStream(ctx context.Context, t *testing.T, uploadID string, uploader *eventstest.MemoryUploader) []apievents.AuditEvent {
 	t.Helper()
 
 	parts, err := uploader.GetParts(uploadID)
 	require.NoError(t, err)
 
-	var outEvents []apievents.AuditEvent
-	var reader *events.ProtoReader
-	for i, part := range parts {
-		if i == 0 {
-			reader = events.NewProtoReader(bytes.NewReader(part))
-		} else {
-			err := reader.Reset(bytes.NewReader(part))
-			require.NoError(t, err)
-		}
-		out, err := reader.ReadAll(ctx)
-		require.NoError(t, err, "part crash %#v", part)
-
-		outEvents = append(outEvents, out...)
+	// combine all uploaded parts to create the session recording content
+	var sessionRecordingContent bytes.Buffer
```

Comment:
For increased confidence, I created a new pull request right off of master to demonstrate this test change + removing `Reset` still passes: #50875